### DISCUSSION TITLE
Confusion About Using Biome Alongside typescript-language-server

### DISCUSSION THREAD

**Discussion by jexjws** [biomejs/biome#3019] [General]
Hello,

I've noticed that even though Biome fully supports LSP, but the documentation mentioned "Now you can use biome alongside typescript-language-server"（ [editors/helix/manual.md](https://github.com/biomejs/biome/blob/main/editors/helix/manual.md) ）. I'm curious to understand the rationale behind this recommendation.

Is it because Biome and typescript-language-server complement each other in some ways? Or are there specific features in typescript-language-server that are not yet available in Biome?

Any advice or insights would be greatly appreciated.

**Comment by ematipico** [comment]
Biome doesn't provide all the features you would expect from a proper language server. We will eventually.
